src/vehicle.cpp

```diff
@@ -6460,9 +6463,8 @@ void vehicle::calc_mass_center( bool use_precalc ) const
             m_part += j.weight();
         }
 
-        if( pi.has_flag( VPFLAG_BOARDABLE ) && parts[i].has_flag( vehicle_part::passenger_flag ) ) {
-            const player *p = get_passenger( i );
-            // Sometimes flag is wrongly set, don't crash!
+        if( pi.has_flag( VPFLAG_BOARDABLE ) ) {
+            const Creature *p = get_passenger( i );
```

Comment:
Btw. (not really needed in this PR): checking the boardable flag *before* calling `get_passenger` is redundant as well. Just call the function, if the target part does not have the flag, the function will return `nullptr` anyway.